### PR TITLE
SALTO-3518 - Upgrade OSS to Node.js v18

### PR DESCRIPTION
**Pull request by tomsellek**
Upgrade to Node.js 18

- [x] Logger colorization tests fail
- [x] Persistent Pool test `when renew throws unknown error should stop silently` fails
- [x] Need to put nexe-patched node18 executables in a network path where nexe can find them

---

---
_Release Notes_: 
N/A

---
_User Notifications_: 
N/A

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -128,6 +128,8 @@ jobs:
           command: yarn test -w 4 --reporters=default --reporters=jest-junit
           environment:
             JEST_JUNIT_OUTPUT: "reports/unittests/test-results.xml"
+            # https://github.com/pinojs/koa-pino-logger/issues/16
+            FORCE_COLOR: 1
```
Comment (amitgoldarb) [inline .circleci/config.yml:132]: Tricky!
Do we have to do this only in Circle? Or does this affect the CLI as well?

**Review by amitgoldarb — APPROVED** [review]
Looks really good!
Great work here!
I'm approving from my POV, added a small question though.
I think we should possibly get another approval from @netama / @alonstern before merging this one though as it mide have some side effects.

File: .circleci/config.yml (hunk 1 above)
Comment (tomsellek) [reply .circleci/config.yml:132]: I couldn't get the tests to fail on my machine or on the CircleCI machine. My theory is that the stream object changed its behaviour when the output is piped, and that CircleCI pipes the output from the tests to a file.

File: packages/cli/package_native.js [hunk 2]
```diff
@@ -16,22 +16,46 @@
...
+ * and the binaries you uploaded should be used automatically.
+ * */
+
+const BUILD_NODE_EXECUTABLE = false
+
 const TARGET_FILE_BASENAME = 'salto'
 const TARGET_DIR = 'pkg'
-const TARGET_ARCH = 'x64'
-const TARGET_NODE_VERSION = '14.15.3'
+const TARGET_ARCH = BUILD_NODE_EXECUTABLE ? os.arch() : 'x64'
```
Comment (alonstern) [inline packages/cli/package_native.js:51]: Why is it always `'x64'` if `BUILD_NODE_EXECUTABLE` is false? does the TARGET_ARCH even affect anything if we are not building nexe?

**Review by alonstern — COMMENTED** [review]
Looks good 👌 added one question

File: packages/cli/package_native.js (hunk 2 above)
Comment (tomsellek) [reply packages/cli/package_native.js:51]: It actually does - it determines which pre-built Node.js binary we use. For the case where we don't build Node.js, the existing code always built for x64, so I left things as they were. Interestingly, this means we never really had a way to build the Salto CLI for M1 Macs. 
We could probably make the target arch part of the different target configs (like we do with OS), but that's a matter for another PR.

**Review by alonstern — APPROVED** [review]
Looks great!

**Pull request merged**